### ISSUE TITLE
Hidden dependency on active_support

### ISSUE THREAD

**Issue by oogali** [thoughtbot/shoulda-matchers#877]
**TL;DR version: Consider changing references from the active_support-specific ```in?``` to the pure Ruby form: ```include?```**

I have a project that uses Sinatra and ActiveRecord, but not Rails. For testing, I'm using RSpec, Factory Girl, and shoulda-matchers.

I'm attempting to write a test to validate the uniqueness of a column with a scope:
```
it { should validate_uniqueness_of(:priority).scoped_to(:publisher_id) }
```

but running into an exception on each run:
```
     Failure/Error: it { should validate_uniqueness_of(:priority).scoped_to(:publisher_id) }
     NoMethodError:
       undefined method `in?' for 5:Fixnum
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:533:in `boolean_value?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:528:in `block (2 levels) in all_scopes_are_booleans?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:528:in `each'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:528:in `all?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:528:in `block in all_scopes_are_booleans?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:527:in `each'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:527:in `all?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:527:in `all_scopes_are_booleans?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:441:in `validate_after_scope_change?'
     # ./gems/gems/shoulda-matchers-3.0.1/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb:266:in `matches?'
     # ./spec/models/app_group_spec.rb:10:in `block (3 levels) in <top (required)>'
```

It appears this is due to  ```Should::Matchers::ActiveRecord::ValidateUniquenessOfMatcher#boolean_value?``` calling the ```in?``` method on the value of the scoped parameter.

```in?``` is a construct of active_support [\[1\]](http://guides.rubyonrails.org/active_support_core_extensions.html#in-questionmark)[\[2\]](https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/core_ext/object/inclusion.rb#L10-L14)[\[3\]](http://stackoverflow.com/questions/28272550/difference-between-in-and-include-in-rails), which is a gem that may not necessarily be included in a non-Rails project.

In this particular instance, would it make sense to change the logic from ```element.in?(array)``` to ```array.include?(element)``` to avoid the Rails/active_support dependency? If so, can it be applied project-wide (I've only spotted 1 other occurrence of this pattern, in ```Shoulda::Matchers::ActionController::RouteParams#symbolize_or_stringify```.

Environment:
- Ruby 2.1.7p400
- RSpec 3.1.7
- shoulda-matchers 3.0.1

1: http://guides.rubyonrails.org/active_support_core_extensions.html#in-questionmark
2: https://github.com/rails/rails/blob/4-2-stable/activesupport/lib/active_support/core_ext/object/inclusion.rb#L10-L14
3: http://stackoverflow.com/questions/28272550/difference-between-in-and-include-in-rails

**Comment by mcmire** [comment]
> Consider changing references from the active_support-specific `in?` to the pure Ruby form: `include?`

Sure, I would be okay with that.

To be fair, ActiveRecord does depend on ActiveSupport internally, and shoulda-matchers also depends explicitly on ActiveSupport 4 too (see the [gemspec](https://github.com/thoughtbot/shoulda-matchers/blob/master/shoulda-matchers.gemspec#L22)). The thing is, of course, that just because you're requiring the active_support gem doesn't mean you're requiring *all* of it, so whatever file provides `in?` probably isn't getting required as well.

Even so, I've never been a big fan of `in?`. So I am fine with changing it to `include?`.

**Comment by oogali** [comment]
That's true, I overlooked that dependency, thanks for pointing that out.

I'll put together a PR this morning.
